server/api/support.py

```diff
@@ -64,6 +64,29 @@ def list_organizations():
     )
 
 
+ORGANIZATION_SCHEMA = {
+    "type": "object",
+    "properties": {"name": {"type": "string"}},
+    "additionalProperties": False,
+    "required": ["name"],
+}
+
+
+@api.route("/support/organizations", methods=["POST"])
+@restrict_access_superadmin
+def create_organization():
+    organization = request.get_json()
+    validate(organization, ORGANIZATION_SCHEMA)
+
+    if Organization.query.filter_by(name=organization["name"]).one_or_none():
+        raise Conflict("Organization already exists")
+
+    db_session.add(Organization(id=str(uuid.uuid4()), name=organization["name"]))
```

Comment:
when there is a DB constraint on uniqueness, I prefer to just `try` and `catch` the constraint exception, because technically doing the if then `add` doesn't fully protect you from the race condition. Not strictly necessary, I can live with this, just highlighting a different, slightly more resilient approach.

Reply:
great point. this has come up before, and i actually initially implemented my first find_or_create in Arlo that way, but I had to do a string match on the constraint name in some exception metadata field, and it seemed so hacky that i stopped doing it

i would love to have a resilient, generalized find_or_create function in Arlo that actually checked for the db exception. but it probably doesn't matter for this particular PR